Microsoft.Toolkit.Uwp.UI.Controls/HamburgerMenu/HamburgerMenu.cs

```diff
@@ -162,14 +164,41 @@ private void NavViewSetItemsSource()
 
                     foreach (var option in options)
                     {
-                        combined.Add(option);
+                        if (!UseNavigationViewSettingsWhenPossible)
+                        {
+                            combined.Add(option);
+                        }
+                        else
+                        {
+                            if (_settingsObject == null && IsSettingsItem(option))
+                            {
+                                _settingsObject = option;
+                                navView.IsSettingsVisible = true;
+                            }
+                            else
+                            {
+                                combined.Add(option);
+                            }
+                        }
                     }
                 }
 
                 navView.MenuItemsSource = combined;
             }
         }
 
+        private bool IsSettingsItem(object menuItem)
+        {
+            Dictionary<string, object> values = menuItem.GetType()
+                                                        .GetProperties()
+                                                        .ToDictionary(p => p.Name, p => p.GetValue(menuItem));
+
+            return menuItem.GetType()
+                           .GetProperties()
+                           .Any(p => p.GetValue(menuItem).ToString().IndexOf("setting", StringComparison.OrdinalIgnoreCase) >= 0
```

Comment:
My assumption is that very few apps have more than one entry in the OptionsItems list and so reversing wouldn't add a lot. It would also make the code for adding non-settings items more complicated.

That this searches for unlocalized text was discussed previously on the issue #1869. The aim of this approach was to try and make it work for the majority of cases with the minimum amount of effort. I wanted adding new properties on the items to explicitly indicate which item to use for settings to be the fallback and not a requirement as this is more work for the developer (they have to do at least two things, not just one) and also, as the item could be potentially anything, it makes coding for this harder.